scheduled-jobs/scanning/art-notify/art-notify.py

```diff
@@ -44,25 +44,28 @@ def get_failed_jobs_text():
                 continue
             job_name = job['name']
             failed_job_ids = []
+            total_eligible_builds = 0
             for build in job['builds']:
                 dt = datetime.fromtimestamp(build['timestamp'] / 1000, tz=timezone.utc)
                 td = now - dt
                 hours = td.seconds // 3600
                 minutes = (td.seconds // 60) % 60
                 if not (td.days == 0 and hours < FAILED_JOB_HOURS):
                     continue
+                # Filter all builds that were not triggered by our automation account
+                # We do not want to report on these since they are manually triggered and
+                # would be monitored by whoever triggered them
+                # Builds which are triggered by another build will not have userId
+                # so we include them by default
+                user_id = next(a['causes'][0].get('userId') for a in build.get('actions') if a.get('causes'))
+                if user_id and user_id != ART_BOT_JENKINS_USERID:
+                    continue
+                total_eligible_builds += 1
                 if build['result'] == 'FAILURE':
-                    # Filter all builds that were not triggered by our automation account
-                    # We do not want to report on these since they are manually triggered and
-                    # would be monitored by whoever triggered them
-                    # Builds which are triggered by another build will not have userId
-                    # so we include them by default
-                    user_id = next(a['causes'][0].get('userId') for a in build.get('actions') if a.get('causes'))
-                    if user_id and user_id != ART_BOT_JENKINS_USERID:
-                        continue
                     failed_job_ids.append(build['number'])
             if len(failed_job_ids) > 0:
-                failed_jobs.append((unquote(job_name), failed_job_ids, job['url']))
+                fail_rate = (len(failed_job_ids)/total_eligible_builds)*100
```

Comment:
Always afraid for `ZeroDivisionError`. Think it is not logically possible, but would test for divisor != 0 or catch

Reply:
Looks like if `failed_job_ids > 0` we should always get `total_eligible_builds > 0`, so this should not happen